src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+                                new QNamesForKmersFinder(broadcastKmerAndIntervalsSet.value())), false)
+                .collect();
+
+        broadcastKmerAndIntervalsSet.destroy();
+
+        return qNames;
+    }
+
+    /** find kmers for each interval */
+    @VisibleForTesting static List<KmerAndInterval> getKmerIntervals(
+            final JavaSparkContext ctx,
+            final HopscotchHashSet<QNameAndInterval> qNamesSet,
+            final Set<SVKmer> kmerKillSet,
+            final JavaRDD<GATKRead> reads,
+            final Locations locations,
+            final PipelineOptions pipelineOptions ) {
+
+        final Broadcast<Set<SVKmer>> broadcastKmerKillSet = ctx.broadcast(kmerKillSet);
+        final Broadcast<HopscotchHashSet<QNameAndInterval>> broadcastQNameAndIntervalsSet =
+                ctx.broadcast(qNamesSet);
+
+        // given a set of template names with interval IDs and a kill set of ubiquitous kmers,
+        // produce a set of interesting kmers for each interval ID
+        final List<KmerAndInterval> kmerIntervals =
+            reads
+            .filter(read -> !read.isDuplicate() && !read.failsVendorQualityCheck() &&
+                    !read.isSecondaryAlignment() && !read.isSupplementaryAlignment())
+            .mapPartitionsToPair(readItr ->
+                    new MapPartitioner<>(readItr,
+                        new QNameKmerizer(broadcastQNameAndIntervalsSet.value(),
+                                        broadcastKmerKillSet.value())), false)
+            .reduceByKey(Integer::sum)
+            .mapPartitions(itr -> new KmerCleaner().call(itr))
+            .collect();
+
+        broadcastQNameAndIntervalsSet.destroy();
+        broadcastKmerKillSet.destroy();
+
+        // record the kmers with their interval IDs
+        if ( locations.kmerFile != null ) {
+            try (final OutputStreamWriter writer = new OutputStreamWriter(new BufferedOutputStream(
+                    BucketUtils.createFile(locations.kmerFile, pipelineOptions)))) {
+                for (final KmerAndInterval kmerAndInterval : kmerIntervals) {
+                    writer.write(kmerAndInterval.toString() + "\n");
+                }
+            } catch (final IOException ioe) {
+                throw new GATKException("Can't write kmer intervals file " + locations.kmerFile, ioe);
+            }
+        }
+
+        return kmerIntervals;
+    }
+
+    @VisibleForTesting static List<SVKmer> getHighCountKmers( final JavaRDD<GATKRead> reads ) {
+        final int nPartitions = reads.partitions().size();
+        return reads
+                .filter(read ->
+                        !read.isSecondaryAlignment() && !read.isSupplementaryAlignment() &&
+                                !read.isDuplicate() && !read.failsVendorQualityCheck())
+                .mapPartitions(KmerCounter::new, false)
+                .mapToPair(kmerAndCount -> new Tuple2<>(new SVKmer(kmerAndCount), kmerAndCount.getCount()))
```

Comment:
Reworked into separate KmerCounter and KmerReducer objects so that the shuffle can use the more economical KmerAndCount class.  Substantially reduces memory use.